prescient/simulator/prescient.py

```diff
@@ -11,127 +11,103 @@
 from .options import Options
 from .simulator import Simulator
 from .data_manager import DataManager
 from .time_manager import TimeManager
 from .oracle_manager import OracleManager
 from .stats_manager import StatsManager
 from .reporting_manager import ReportingManager
+from prescient.scripts import runner
 from prescient.stats.overall_stats import OverallStats
-import prescient.plugins 
-
 from prescient.engine.egret import EgretEngine as Engine
 
-try:
-    import pstats
-    pstats_available=True
-except ImportError:
-    pstats_available=False
-
-def create_prescient(options: Options):
-    engine = Engine()
-    time_manager = TimeManager()
-    data_manager = DataManager()
-    oracle_manager = OracleManager()
-    stats_manager = StatsManager()
-    reporting_manager = ReportingManager()
-    prescient = Simulator(engine, time_manager, data_manager, oracle_manager, stats_manager, reporting_manager)
-    return prescient
-
-
-def main_prescient(options: Options):
-    ans = None
-
-    if pstats_available and options.profile > 0:
-        #
-        # Call the main ef writer with profiling.
-        #
-        tfile = pyutilib.services.TempfileManager.create_tempfile(suffix=".profile")
-        tmp = profile.runctx('simulate(options)', globals(), locals(), tfile)
-        p = pstats.Stats(tfile).strip_dirs()
-        p.sort_stats('time', 'cumulative')
-        p = p.print_stats(options.profile)
-        p.print_callers(options.profile)
-        p.print_callees(options.profile)
-        p = p.sort_stats('cumulative', 'calls')
-        p.print_stats(options.profile)
-        p.print_callers(options.profile)
-        p.print_callees(options.profile)
-        p = p.sort_stats('calls')
-        p.print_stats(options.profile)
-        p.print_callers(options.profile)
-        p.print_callees(options.profile)
-        pyutilib.services.TempfileManager.clear_tempfiles()
-        ans = [tmp, None]
-
-    else:
-        if options.traceback is True:
-            simulator = create_prescient(options)
-            ans = simulator.simulate(options)
+from pyutilib.misc import import_file
+
+class Prescient(Simulator):
+
+    CONFIG = PrescientConfig
+
+    def __init__(self):
+
+        engine = Engine()
+        time_manager = TimeManager()
+        data_manager = DataManager()
+        oracle_manager = OracleManager()
+        stats_manager = StatsManager()
+        reporting_manager = ReportingManager()
+
+        self.simulate_called = False
+
+        super().__init__(engine, time_manager, data_manager, oracle_manager, stats_manager, reporting_manager)
+
+    def simulate(self, **options):
+        # coming from Python
+        # For safety, in case we re-run Prescient
+        # again in the same Python script
+        # (For the same reason, we don't accept options
+        #  in __init__ above.)
+        prescient.plugins.internal.clear_plugins()
+        prescient.simulator.config.clear_prescient_config()
```

Comment:
Personally I'd prefer that we move away from a global config. In the current code it isn't obvious when settings persist and when they get reset.  For example, what's the mipgap if I do this?
```
pres = Prescient()
pres.CONFIG.ruc_mipgap = .05
pres.simulate()
```
The mipgap for this run will be the default 0.01.

My preference would be a per-instance config instead of a global one.

Reply:
Darryl has added a per-instance config. I think for Prescient this is the safest thing to do.